cni-plugin/test/install-cni_test.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2018 Istio Authors
```

Comment:
What the recommend way to run this test file? Locally, I get:

```
go test ./cni-plugin/test
--- FAIL: TestInstallCNI_Scenario1 (1.63s)
    install-cni_test.go:254: If the test fails, you will want to check the docker logs of the container and then be sure to stop && remove it before running the tests again.
    install-cni_test.go:256: Scenario 1: There isn't an existing plugin configuration in the CNI_NET_DIR.
    install-cni_test.go:257: GIVEN the CNI_NET_DIR=/etc/cni/net.d/ is empty
...
```

When we run tests in CI, we use `go test ./...`, so if this test can't pass on a standard run, we'll need to disable it by default, and then include instructions in TEST.md for running it. We already have a few integration tests that behave like that, and they use a `TestMain` func to require an additional flag in order to start the tests. For instance:

https://github.com/linkerd/linkerd2/blob/92f2cd9b639cfe694837d83c0e1e7111826aa5b4/proxy-init/integration_test/iptables/http_test.go#L20-L30

Reply:
I just cd into the cni-plugin folder and run go test ./... and it works on my box. My colleague is having similar issues though so I will investigate a bit more.

Reply:
Any chance you can provide more of the error you are seeing?

Reply:
I did forget that you need to make sure the TAG env is correct. `TAG=dev-asdf-user go test ./...`

Reply:
Ah, hmm, it looks like maybe the 3 tests are running in parallel and conflicting with one another? I see:

```
?   	github.com/linkerd/linkerd2/cni-plugin	[no test files]
--- FAIL: TestInstallCNI_Scenario1 (1.10s)
    install-cni_test.go:254: If the test fails, you will want to check the docker logs of the container and then be sure to stop && remove it before running the tests again.
...
    install-cni_test.go:156: Test 1 ERROR: failed to start docker container 'gcr.io/linkerd-io/cni-plugin:git-7a65156b', see /Users/kl/workspace/go/src/github.com/linkerd/linkerd2/cni-plugin/test//docker_run_stderr
--- FAIL: TestInstallCNI_Scenario2 (0.06s)
    install-cni_test.go:282: If the test fails, you will want to check the docker logs of the container and then be sure to stop && remove it before running the tests again.
...
    install-cni_test.go:156: Test 2 ERROR: failed to start docker container 'gcr.io/linkerd-io/cni-plugin:git-7a65156b', see /Users/kl/workspace/go/src/github.com/linkerd/linkerd2/cni-plugin/test//docker_run_stderr
--- FAIL: TestInstallCNI_Scenario3 (0.07s)
    install-cni_test.go:313: If the test fails, you will want to check the docker logs of the container and then be sure to stop && remove it before running the tests again.
...
    install-cni_test.go:156: Test 3 ERROR: failed to start docker container 'gcr.io/linkerd-io/cni-plugin:git-7a65156b', see /Users/kl/workspace/go/src/github.com/linkerd/linkerd2/cni-plugin/test//docker_run_stderr
FAIL
FAIL	github.com/linkerd/linkerd2/cni-plugin/test	1.249s
```

And when I cat that file, I see:

```
docker: Error response from daemon: Conflict. The container name "/test-linkerd-cni-install" is already in use by container "b5ae08f93ed7996106b6e9ac71506ee5ac733038dedbe49998a4fc05097a22e2". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
```

Reply:
Typically, what happens is that when there is a failure, the test doesn't stop the previous docker container. I'm trying to figure out if it should always try to shutdown the test container or maybe I use a different container name per test. Do you know if each test run, travis-ci provides a clean docker environment where there aren't any running containers?